### PR TITLE
feat: support the `score` method for `PaLM2TextGenerator`

### PR DESCRIPTION
**Pull request by ashleyxuu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes internal #334660953 🦕

### CODE REVIEW

File: tests/system/load/test_llm.py [hunk 1]
```diff
@@ -69,3 +68,21 @@ def test_llm_palm_configure_fit(llm_fine_tune_df_default_index, llm_remote_text_
     assert all(series.str.len() == 1)
 
     # TODO(ashleyxu b/335492787): After bqml rolled out version control: save, load, check parameters to ensure configuration was kept
+
+
+def test_llm_palm_score(llm_fine_tune_df_default_index):
```
Comment (GarrettWu) [inline tests/system/load/test_llm.py:73]: We may want to add at least a test with argument.

File: bigframes/ml/ensemble.py [hunk 2]
```diff
@@ -472,7 +472,7 @@ def predict(
     def score(
         self,
         X: Union[bpd.DataFrame, bpd.Series],
-        y: Union[bpd.DataFrame, bpd.Series],
+        y=None,  # ignored
```
Comment (GarrettWu) [inline bigframes/ml/ensemble.py:475]: Why is it?

File: bigframes/ml/llm.py [hunk 3]
```diff
@@ -310,6 +310,64 @@ def predict(
 
         return df
 
+    def score(
+        self,
+        X: Union[bpd.DataFrame, bpd.Series],
+        y=None,  # ignored
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:316]: After a second thought, we actually want input_text as X and output_text as y.

File: bigframes/ml/llm.py [hunk 4]
```diff
@@ -310,6 +310,64 @@ def predict(
...
+        task_type: Literal[
+            "TEXT_GENERATION", "CLASSIFICATION", "SUMMARIZATION", "QUESTION_ANSWERING"
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:318]: Lower cases to be consistent.

Comment (ashleyxuu) [reply bigframes/ml/llm.py]: done

File: bigframes/ml/llm.py (hunk 3 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py]: Done

File: bigframes/ml/ensemble.py (hunk 2 above)
Comment (ashleyxuu) [reply bigframes/ml/ensemble.py]: My mistake. Revert it back.

File: tests/system/load/test_llm.py (hunk 1 above)
Comment (ashleyxuu) [reply tests/system/load/test_llm.py:73]: Done.

File: bigframes/ml/sql.py [hunk 5]
```diff
@@ -318,6 +318,20 @@ def ml_evaluate(self, source_df: Optional[bpd.DataFrame] = None) -> str:
             return f"""SELECT * FROM ML.EVALUATE(MODEL `{self._model_name}`,
   ({source_sql}))"""
 
+    # ML evaluation TVFs
+    def ml_llm_evaluate(
+        self, source_df: Optional[bpd.DataFrame] = None, task_type: Optional[str] = None
+    ) -> str:
+        """Encode ML.EVALUATE for BQML"""
+        if source_df is None:
```
Comment (GarrettWu) [inline bigframes/ml/sql.py:326]: Looks like source_df can't be None. Otherwise it can't be processed.

Comment (ashleyxuu) [reply bigframes/ml/sql.py]: Done.

File: bigframes/ml/llm.py [hunk 6]
```diff
@@ -310,6 +310,63 @@ def predict(
...
+            for the outputs relevant to this model type.
+
+        Args:
+            X (bigframes.dataframe.DataFrame or bigframes.series.Series):
+                A BigQuery DataFrame as evaluation data, which contains only one column of input_text
+                that contains the prompt text to use when evaluating the model.
+            y (bigframes.dataframe.DataFrame or bigframes.series.Series):
+                A BigQuery DataFrame as evaluation labels, which contains only one column of output_text
+                that you would expect to be returned by the model.
+            task_type (Optional[str]):
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:343]: Not an optional[str], but str with default.

**Review by GarrettWu — APPROVED** [review]

File: bigframes/ml/llm.py (hunk 6 above)
Comment (ashleyxuu) [reply bigframes/ml/llm.py]: Done.

**Pull request merged**